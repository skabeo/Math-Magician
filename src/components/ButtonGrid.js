import React from 'react';
import PropTypes from 'prop-types';

const getClass = (index) => {
  if (index === 16) {
    return 'two-spaces';
  }
  if (index === 3 || index === 7 || index === 11 || index === 15 || index === 18) {
    return 'orange';
  }
  return '';
};

const ButtonGrid = (props) => {
  const { buttons } = props;
  return (
    <div className="button-grid">
      {buttons.map((button, index) => (
        <button type="button" key={button} className={getClass(index)}>{button}</button>
      ))}
    </div>
  );
};

ButtonGrid.propTypes = {
  buttons: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default ButtonGrid;
