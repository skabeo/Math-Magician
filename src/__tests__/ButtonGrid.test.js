import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ButtonGrid from '../components/ButtonGrid';

describe('ButtonGrid component', () => {
  test('Calculator buttons work as expected', () => {
    const { getByPlaceholderText, getByText } = render(<ButtonGrid />);
    const output = getByPlaceholderText('0');

    fireEvent.click(getByText('7'));
    expect(output.value).toBe('7');

    fireEvent.click(getByText('x'));
    expect(output.value).toBe('7');

    fireEvent.click(getByText('5'));
    expect(output.value).toBe('5');

    fireEvent.click(getByText('='));
    expect(output.value).toBe('35');
  });

  it('should render correctly', () => {
    const { asFragment } = render(<ButtonGrid />);
    expect(asFragment()).toMatchSnapshot();
  });
});
