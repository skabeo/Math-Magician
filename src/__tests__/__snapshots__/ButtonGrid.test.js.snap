// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGrid component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="calculator-grid"
  >
    <input
      class="output"
      dir="rtl"
      placeholder="0"
    />
    <button
      type="button"
    >
      AC
    </button>
    <button
      type="button"
    >
      +/-
    </button>
    <button
      type="button"
    >
      %
    </button>
    <button
      class="orange"
      type="button"
    >
      รท
    </button>
    <button
      type="button"
    >
      7
    </button>
    <button
      type="button"
    >
      8
    </button>
    <button
      type="button"
    >
      9
    </button>
    <button
      class="orange"
      type="button"
    >
      x
    </button>
    <button
      type="button"
    >
      4
    </button>
    <button
      type="button"
    >
      5
    </button>
    <button
      type="button"
    >
      6
    </button>
    <button
      class="orange"
      type="button"
    >
      -
    </button>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      class="orange"
      type="button"
    >
      +
    </button>
    <button
      class="two-spaces"
      type="button"
    >
      0
    </button>
    <button
      type="button"
    >
      .
    </button>
    <button
      class="orange"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
